crates/ruff_linter/src/rules/flake8_use_pathlib/rules/dotless_pathlib_with_suffix.rs

```diff
@@ -75,18 +78,30 @@ pub(crate) fn dotless_pathlib_with_suffix(checker: &mut Checker, call: &ast::Exp
 
     let string_value = string.value.to_str();
 
-    if string_value.is_empty() || string_value.starts_with('.') {
+    if string_value.is_empty() {
+        return;
+    }
+
+    if string_value.starts_with('.') && string_value.len() > 1 {
         return;
     }
 
     let [first_part, ..] = string.value.as_slice() else {
         return;
     };
 
-    let diagnostic = Diagnostic::new(DotlessPathlibWithSuffix, call.range);
-    let after_leading_quote = string.start() + first_part.flags.opener_len();
-    let fix = Fix::unsafe_edit(Edit::insertion(".".to_string(), after_leading_quote));
-    checker.diagnostics.push(diagnostic.with_fix(fix));
+    let mut diagnostic = Diagnostic::new(DotlessPathlibWithSuffix, call.range);
+
+    diagnostic.try_set_fix(|| {
+        if string_value == "." {
+            return Err(anyhow::anyhow!("Cannot fix the suffix `.`"));
+        }
+        let after_leading_quote = string.start() + first_part.flags.opener_len();
+        let fix = Fix::unsafe_edit(Edit::insertion(".".to_string(), after_leading_quote));
+        Ok(fix)
+    });
```

Comment:
I'd prefer avoiding `try_set_fix` when the check when to set a diagnostic is fairly trivial:

```suggestion
		if string_value != "." {
      let after_leading_quote = string.start() + first_part.flags.opener_len();
	    diagnostic.set_fix(Fix::unsafe_edit(Edit::insertion(".".to_string(), after_leading_quote)));
    }
```

Reply:
Thank you. This is helpful.

Yesterday I spent time to write this part of the function in rust. I had not understood the semantic of `try_to_fix`. I thought it was necessary to specify an `Err` if there is no fix.